### DISCUSSION TITLE
Powershell

### DISCUSSION THREAD

**Discussion by satyoti** [PowerShell/PowerShell#23787] [Q&A]
How to learn powershell without any programing language. When I am trying to learn .I am feeling it is difficult to learn but I want to learn.please suggest

**Comment by rhubarb-geek-nz** [comment]
Hi, welcome to PowerShell. If you have no programming experience then you need to start by understanding a few basic concepts such as variables and flow control, that applies to almost any language.

When learning any new language, I always start with what is called a "Hello World" program. Every language has one and is just a simple program that prints a string.

https://learn.microsoft.com/en-us/training/modules/powershell-write-first/

Once you are happy with a simple program, set yourself a simple task which is meaningful to you and you have an interest in getting it completed. Then start by breaking that task down to short sequences of code and try writing code that achieves each step. Then slowly put all the pieces together to make the complete program.

Learning to Program is like learning to Play a Guitar. You will make many mistakes along the way but that is part of the adventure.
